be/src/olap/reader.cpp

```diff
@@ -503,43 +485,93 @@ OLAPStatus Reader::_capture_rs_readers(const ReaderParams& read_params) {
+                        << ", end_key=" << end_key->to_string();
+                eof = true;
+                break;
+            }
+            is_lower_key_included = true;
+        } else if (_keys_param.range.compare("ge") == 0) {
+            if (end_key != nullptr && start_key->cmp(*end_key) > 0) {
+                VLOG(3) << "return EOF when range=" << _keys_param.range
+                        << ", start_key=" << start_key->to_string()
+                        << ", end_key=" << end_key->to_string();
+                eof = true;
+                break;
+            }
+            is_lower_key_included = false;
+        } else if (0 == _keys_param.range.compare("eq")) {
+            is_lower_key_included = false;
+            is_upper_key_included = true;
         } else {
-            VLOG(3) << "not filter delta in delete predicate:"
-                    << i_data->version().first << ", " << i_data->version().second;
-            i_data->set_delete_status(DEL_NOT_SATISFIED);
+            LOG(WARNING) << "reader params range is error. "
+                         << "range=" << _keys_param.to_string();
+            return OLAP_ERR_READER_GET_ITERATOR_ERROR;
+        }
+        _is_lower_keys_included.push_back(is_lower_key_included);
+        _is_upper_keys_included.push_back(is_upper_key_included);
+    }
+
+    if (eof) { return OLAP_SUCCESS; }
+
+    RowsetReaderContextBuilder context_builder;
+    context_builder.set_tablet_schema(&_tablet->tablet_schema())
+                   .set_return_columns(&_return_columns)
+                   .set_load_bf_columns(&_load_bf_columns)
+                   .set_conditions(&_conditions)
+                   .set_predicates(&_col_predicates)
+                   .set_lower_bound_keys(&_keys_param.start_keys)
+                   .set_is_lower_keys_included(&_is_lower_keys_included)
+                   .set_upper_bound_keys(&_keys_param.end_keys)
+                   .set_is_upper_keys_included(&_is_upper_keys_included)
+                   .set_delete_handler(&_delete_handler)
+                   .set_stats(&_stats)
+                   .set_is_using_cache(is_using_cache)
+                   .set_runtime_state(read_params.runtime_state);
+    ReaderContext context = context_builder.build();
+    for (auto& rs_reader : *rs_readers) {
+        rs_reader->init(&context);
+        _rs_readers.push_back(rs_reader);
+    }
+
+    for (auto& rs_reader : _rs_readers) {
+        RowBlock* block = nullptr;
+        auto res = rs_reader->next_block(&block);
+        if (res == OLAP_SUCCESS) {
+            res = _collect_iter->add_child(rs_reader, block);
+            if (res != OLAP_SUCCESS && res != OLAP_ERR_DATA_EOF) {
+                LOG(WARNING) << "failed to add child to iterator";
+                return res;
+            } else if (res == OLAP_ERR_DATA_EOF) {
+                continue;
+            } else {
+                LOG(WARNING) << "prepare block failed, res=" << res;
```

Comment:
I think there is a bug here. 'else' here stands for OLAP_SUCCESS

Reply:
OK, I will fixed it.